src/lib/mongoBackend/mongoRegistrationGet.cpp

```diff
@@ -0,0 +1,325 @@
+* possible in V2 and we cannot respond to the request using the current implementation of V2
+*
+*/
+static bool setDataProvided(ngsiv2::Registration* regP, const mongo::BSONObj& r, bool arrayAllowed)
+{
+  std::vector<mongo::BSONElement> crV = getFieldF(r, REG_CONTEXT_REGISTRATION).Array();
+
+  if (crV.size() > 1)
+  {
+    return false;
+  }
+
+  //
+  // Extract first CR from the contextRegistration vector
+  //
+  mongo::BSONObj cr0 = crV[0].embeddedObject();
+
+
+
+  //
+  // Entities
+  //
+  std::vector<mongo::BSONElement>  dbEntityV = getFieldF(cr0, REG_ENTITIES).Array();
+
+  for (unsigned int ix = 0; ix < dbEntityV.size(); ++ix)
+  {
+    ngsiv2::EntID    entity;
+    mongo::BSONObj   ce = dbEntityV[ix].embeddedObject();
+
+    if (ce.hasField(REG_ENTITY_ISPATTERN))
+    {
+      std::string isPattern = getStringFieldF(ce, REG_ENTITY_ISPATTERN);
+
+      if (isPattern == "true")
+      {
+        entity.idPattern = getStringFieldF(ce, REG_ENTITY_ID);
+      }
+      else
+      {
+        entity.id = getStringFieldF(ce, REG_ENTITY_ID);
+      }
+    }
+    else
+    {
+      entity.id = getStringFieldF(ce, REG_ENTITY_ID);
+    }
+    
+    if (ce.hasField(REG_ENTITY_ISTYPEPATTERN))
+    {
+      std::string isPattern = getStringFieldF(ce, REG_ENTITY_ISTYPEPATTERN);
+
+      if (isPattern == "true")
+      {
+        entity.typePattern = getStringFieldF(ce, REG_ENTITY_TYPE);
+      }
+      else
+      {
+        entity.type = getStringFieldF(ce, REG_ENTITY_TYPE);
+      }
+    }
+    else
+    {
+      entity.type = getStringFieldF(ce, REG_ENTITY_TYPE);
+    }
+
+    regP->dataProvided.entities.push_back(entity);
+  }
+
+
+  //
+  // Attributes
+  //
+  std::vector<mongo::BSONElement> dbAttributeV = getFieldF(cr0, REG_ATTRS).Array();
+  for (unsigned int ix = 0; ix < dbAttributeV.size(); ++ix)
+  {
+    mongo::BSONObj  aobj     = dbAttributeV[ix].embeddedObject();
+    std::string     attrName = getStringFieldF(aobj, REG_ATTRS_NAME);
+
+    if (attrName != "")
+    {
+      regP->dataProvided.attributes.push_back(attrName);
+    }
+  }
+
+  //
+  // Providing application
+  //
+  setProvider(regP, cr0);
+
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* setExpires -
+*/
+static void setExpires(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+  regP->expires = (r.hasField(REG_EXPIRATION))? getIntFieldF(r, REG_EXPIRATION) : -1;
+}
+
+
+
+/* ****************************************************************************
+*
+* setStatus -
+*/
+static void setStatus(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+  regP->status = (r.hasField(REG_STATUS))? getStringFieldF(r, REG_STATUS): "";
+}
+
+
+
+/* ****************************************************************************
+*
+* setForwardingInformation -
+*/
+static void setForwardingInformation(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+}
```

Comment:
I'd suggest to add some brief explanation about why this function is empty, even a reference to issue if we have the explanation already in some existing issue.

Reply:
Can't find an issue about API v2 forwarding.
There should be one, right?

Reply:
Fixed in f114c3e (but without issue)

Reply:
> There should be one, right?

I think currently it isn't any issue about that. As much, some early dicussion in an old PR starting here: https://github.com/telefonicaid/fiware-orion/pull/2990#issuecomment-334726286. That could be reused to create the issue.

Any volunteer to create the issue? @jmcanterafonseca? ;)